src/types/authentication.ts

```diff
@@ -0,0 +1,12 @@
+import type { Session, User } from 'next-auth';
+
+export type AuthenticationState =
+  | {
+      user: undefined;
+      status: 'loading';
+    }
+  | {
+      update: (data?: User) => Promise<Session | null>;
+      user: User;
+      status: 'authenticated';
+    };
```

Comment:
ask;

타입을 UnAuthentication와 Authentication를 구분해서 작성해서 적용하는 것 어떠세요?

```suggestion
export interface UnAuthentication {
  user: undefined;
  status: 'loading' | 'unauthenticated';
}

export interface Authentication {
  update: (data?: User) => Promise<Session | null>;
  user: User;
  status: 'authenticated';
};

// useAuthenticationState.ts
export const useAuthenticationState = () => {
...
  const [authenticationState, setAuthenticationState] =
    useState<UnAuthentication | Authentication>({
      user: undefined,
      status: 'loading',
    });
 ...
}
```

Reply:
useAuthentication 훅 사용 시 인증되지 않는(unauthenticated) 경우 메인 페이지로 리다이렉션 처리가 되어 실제로 반환되는 status 타입은 loading과 authenticated 두가지의 경우 밖에 없습니다!

위의 두가지의 경우가 각각 타입을 재활용할 수 있다면 분리해도 좋을 것 같지만, 해당 타입은 재활용할 일이 없어보여 하나로 처리하였습니다!